_lp UDFs do not appear to work

The `_lp` UDFs don't appear much used but I'm about to use them **a lot** :). I was testing this and it produced the following error. This contradicts the manual that says `_lp` functions can access `+= target`. See https://mc-stan.org/docs/2_26/stan-users-guide/functions-accessing-the-log-probability-accumulator.html

```
functions {
  void lprob_lp(real x) {
    target +=  normal_lpdf(x | 0, 1);
  }
}
parameters {
  real x;
}
transformed parameters {
  real ll;
  
  lprob_lp(x);
}
model {
}
```
Generates the error
```
> mod <- cmdstan_model(fp)
Compiling Stan program...
/
Semantic error in '/var/folders/30/ddxkfs1x7bscp4_99c4db5jc0000gn/T/RtmpzPOyNJ/model-1b486a481fa.stan', line 12, column 2 to column 14:
   -------------------------------------------------
    10:    real ll;
    11:    
    12:    lprob_lp(x);
           ^
    13:  }
    14:  model {
   -------------------------------------------------

Target can only be accessed in the model block or in definitions of functions with the suffix _lp.

make: *** [/var/folders/30/ddxkfs1x7bscp4_99c4db5jc0000gn/T/RtmpzPOyNJ/model-1b486a481fa.hpp] Error 1
Error: An error occured during compilation! See the message above for more information.
```

--- a/src/frontend/Semantic_check.ml
+++ b/src/frontend/Semantic_check.ml
@@ -947,7 +947,9 @@ let semantic_check_nrfn_target ~loc ~cf id =
   Validate.(
     if
       String.is_suffix id.name ~suffix:"_lp"
-      && not (cf.in_lp_fun_def || cf.current_block = Model)
+      && not
+           ( cf.in_lp_fun_def || cf.current_block = Model
+           || cf.current_block = TParam )
     then Semantic_error.target_plusequals_outisde_model_or_logprob loc |> error
     else ok ())
 

--- a/test/integration/good/lp_transformed_param.stan
+++ b/test/integration/good/lp_transformed_param.stan
@@ -3,12 +3,16 @@ functions {
     target += normal_lpdf(r | 0, 1);
     return r;
   }
+  void nr_test_lp(real r) {
+    target += normal_lpdf(r | 0, 1);
+  }
 }
 parameters { 
   real y; 
 }
 transformed parameters { 
   real alpha = test_lp(5.0);
+  nr_test_lp(5.0);
 }
 model {
   y ~ normal(0, 1); 

--- a/test/integration/good/pretty.expected
+++ b/test/integration/good/pretty.expected
@@ -3772,12 +3772,16 @@ functions {
     target += normal_lpdf(r| 0, 1);
     return r;
   }
+  void nr_test_lp(real r) {
+    target += normal_lpdf(r| 0, 1);
+  }
 }
 parameters {
   real y;
 }
 transformed parameters {
   real alpha = test_lp(5.0);
+  nr_test_lp(5.0);
 }
 model {
   y ~ normal(0, 1);

